### PR TITLE
fix: Build ubuntu image and use them instead of alpine

### LINKED ISSUE #51: Use of alpine-built static bpftrace is prone to crashing
As described in https://github.com/iovisor/bpftrace/issues/266, it appears that linking to musl instead of glibc is what causes sporadic crashes (deterministically) when loading certain scripts.

For instance, when I run kubectl-trace using `tcpconnect.bt` from bpftrace/tools, it works fine. But, if I try and load `tcpaccept.bt`, I get a crash every time:

```
libclang: crash detected during parsing: {
  'source_filename' : 'definitions.h'
  'command_line_args' : ['clang', '-I', '/bpftrace/include', '-nostdinc', '-isystem', '/virtual/lib/clang/include', '-I/lib/modules/4.14.65+/build/arch/x86/include', '-I/lib/modules/4.14.65+/build/arch/x86/include/generated/uapi', '-I/lib/modules/4.14.65+/build/arch/x86/include/generated', '-I/lib/modules/4.14.65+/build/include', '-I/lib/modules/4.14.65+/build/./arch/x86/include/uapi', '-I/lib/modules/4.14.65+/build/arch/x86/include/generated/uapi', '-I/lib/modules/4.14.65+/build/include/uapi', '-I/lib/modules/4.14.65+/build/include/generated', '-I/lib/modules/4.14.65+/build/include/generated/uapi', '-I./arch/x86/include', '-Iarch/x86/include/generated/uapi', '-Iarch/x86/include/generated', '-Iinclude', '-I./arch/x86/include/uapi', '-Iarch/x86/include/generated/uapi', '-I./include/uapi', '-Iinclude/generated/uapi', '-include', './include/linux/kconfig.h', '-D__KERNEL__', '-D__HAVE_BUILTIN_BSWAP16__', '-D__HAVE_BUILTIN_BSWAP32__', '-D__HAVE_BUILTIN_BSWAP64__', '-Wno-unused-value', '-Wno-pointer-sign', '-fno-stack-protector'],
  'unsaved_files' : [('definitions.h', '...', 22), ('/bpftrace/include/__stddef_max_align_t.h', '...', 1771), ('/bpftrace/include/float.h', '...', 5192), ('/bpftrace/include/limits.h', '...', 3735), ('/bpftrace/include/stdarg.h', '...', 2025), ('/bpftrace/include/stddef.h', '...', 4499), ('/bpftrace/include/stdint.h', '...', 23388)],
  'options' : 0,
}
Clang error while parsing C definitions: 2
Input (22): #include <net/sock.h>

Unknown struct/union: 'sock'
exit status 1
```

Which is consistent with the error and failure mode of https://github.com/iovisor/bpftrace/issues/266.

To fix this, we should either use an ubuntu-based bpftrace that is dynamically linked, or if bpftrace figures out to build a statically linked bpftrace with ubuntu (currently ubuntu doesn't patch a static libclang.a), then use that instead. Of course, maybe someone will figure out a way to make the alpine build work, but I think the issue is in musl so I'm not particularly hopeful.

Until this is fixed, some scripts will randomly crash which makes this tool pretty unreliable.

### PR DESCRIPTION
**Pull request by dalehamel**
Fix for #51 and also bumps bpftrace to 0.9.2 from ubuntu eaon

I was able to confirm that the script that doesn't work with static bpftrace works with the dynamically linked one. This is at the expense of a slightly larger (well, 3x larger), though the size here shouldn't matter *too* much.

As I state in #51, the preferred solution would be to fix the static alpine build (no idea how to do this), or get a static build for ubuntu working (should be possible, others are working on it).

This could at least be a stop-gap until there's a working/non-crashy static image available

### CODE REVIEW

**Comment by dalehamel** [comment]
Looks like I broke CI with this ,will need to see what's up later (train wifi is garbage).

Testing this out locally and seems to work. My approach was to leave the alpine build, and use dependencies to target which build should actually be done in the makefile. This should allow for keeping the alpine image up to date easily.

I moved all the dockerfiles to the build directory as well.

File: hack/ci-build-image.sh [hunk 1]
```diff
@@ -8,5 +8,6 @@ if [[ ! -z "$TRAVIS_PULL_REQUEST_BRANCH" ]]; then
   makeopts="-e GIT_BRANCH=$TRAVIS_PULL_REQUEST_BRANCH"
 fi
 
+$make $makeopts bpftraceimage/build # HACK since the ubuntu image isn't pushed yet
```
Comment (dalehamel) [inline hack/ci-build-image.sh]: It seems this was necessary to get the CI to have the tag locally in its cache, since there is nothing for it to pull.

**Comment by dalehamel** [comment]
The commit history is ugly, i'll rebase before merging if the changes are approved.

File: hack/ci-build-image.sh (hunk 1 above)
Comment (fntlnz) [reply hack/ci-build-image.sh]: Just pushed `quay.io/iovisor/kubectl-trace-bpftrace-base-ubuntu:49ff42b4e311794cddad24a848b3c225a1b61804"` you can remove this now

**Comment by fntlnz** [comment]
Assigned to me to continue work:

Tasks:
- [ ] pin the bpftrace version on this ubuntu image
- [ ] rebase everything
- [ ] make sure we ship at least bpftrace 0.9.2

**Comment by fntlnz** [comment]
@leodido  PTAL

**Comment by dalehamel** [comment]
Thanks guys!

**Pull request merged**